sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +194,696 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
+			s.Assert().Equal(tCase.expectedStatus.rotten, tCase.endpointPair.status.rotten)
+			s.Assert().Equal(tCase.expectedStatus.used, tCase.endpointPair.status.used)
+			if tCase.expectedEndpoint != nil {
+				_, ok := tCase.enrichedConnections[*tCase.expectedEndpoint]
+				s.Assert().True(ok)
+			}
+		})
+	}
+}
+
+func (s *NetworkFlowManagerTestSuite) TestEnrichProcessListening() {
+	mockCtrl := gomock.NewController(s.T())
+	m, mockEntityStore, _, _ := createManager(mockCtrl)
+	deploymentID := "deployment-id"
+	podID := "pod-id"
+	cases := map[string]struct {
+		containerPair               *containerPair
+		enrichedConnections         map[processListeningIndicator]timestamp.MicroTS
+		expectEntityLookupContainer expectFn
+		expectedStatus              *connStatus
+		expectedListeningIndicator  *processListeningIndicator
+	}{
+		"Rotten connection should return rotten status": {
+			containerPair:               createContainerPair(timestamp.Now().Add(-maxContainerResolutionWaitPeriod * 2)),
+			expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{}, false),
+			expectedStatus: &connStatus{
+				rotten:      true,
+				usedProcess: true,
+			},
+		},
+		"Container endpoint should return a processListeningIndicator with the correct id": {
+			containerPair:       createContainerPair(timestamp.Now()),
+			enrichedConnections: make(map[processListeningIndicator]timestamp.MicroTS),
+			expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{
+				DeploymentID:  deploymentID,
+				ContainerName: "container-name",
+				PodID:         podID,
+			}, true),
+			expectedStatus: &connStatus{
+				usedProcess: true,
+			},
+			expectedListeningIndicator: &processListeningIndicator{
+				key: processUniqueKey{
+					podID:         podID,
+					containerName: "container-name",
+					deploymentID:  deploymentID,
+					process:       defaultProcessKey(),
+				},
+				port:     80,
+				protocol: net.TCP.ToProtobuf(),
+			},
+		},
+	}
+	for name, tCase := range cases {
+		s.Run(name, func() {
+			tCase.expectEntityLookupContainer.runIfSet()
+			m.enrichProcessListening(tCase.containerPair.endpoint, tCase.containerPair.status, tCase.enrichedConnections)
+			s.Assert().Equal(tCase.expectedStatus.rotten, tCase.containerPair.status.rotten)
+			s.Assert().Equal(tCase.expectedStatus.usedProcess, tCase.containerPair.status.usedProcess)
+			if tCase.expectedListeningIndicator != nil {
+				_, ok := tCase.enrichedConnections[*tCase.expectedListeningIndicator]
+				s.Assert().True(ok)
+			}
+		})
+	}
+}
+
+func (s *NetworkFlowManagerTestSuite) TestManagerOfflineMode() {
+	s.T().Setenv(env.ProcessesListeningOnPort.EnvVar(), "false")
+	srcID := "src-id"
+	dstID := "dst-id"
+	mockCtrl := gomock.NewController(s.T())
+	hostname := "hostname"
+	containerID := "container-id"
+	m, mockEntity, _, mockDetector := createManager(mockCtrl)
+	states := []struct {
```

Comment:
It would be cool to add names to the test cases, so that one can easier understand what is being tested

```suggestion
	states := map[string]struct {
```

Reply:
This actually needs to be a slice to maintain order. We rely on the connections being added in order to later check for them in the next state.

Reply:
Good point! Maybe we should write this in the comment that the order is important?
We could still add a `name string` to the struct 😉

Reply:
That I can do 🙁

Reply:
Done

Reply:
That looks great, thanks!